File: Dddml.Wms.Common/Specialization/PropertyCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Dddml.Wms.Specialization
{
    public class PropertyCommand<TContent, TState> : IPropertyCommand<TContent, TState>
    {
        public TContent Content { get; set; }

        public Func<TState> GetState { get; set; }

        public Action<TState> SetState { get; set; }

        public string OuterCommandType { get; set; }
    }
}

File: Dddml.Wms.JavaCommon/src/org/dddml/wms/specialization/PropertyCommand.java
package org.dddml.wms.specialization;

import java.util.function.Supplier;
import java.util.function.Consumer;

/**
 * Created by Yang on 2016/7/25.
 */
public interface PropertyCommand <TContent, TState>
{
    TContent getContent();

    void getContent(TContent content);

    Supplier<TState> getStateGetter();

    void setStateGetter(Supplier<TState> stateGetter);

    Consumer<TState> getStateSetter();

    void getStateSetter(Consumer<TState> stateSetter);

    String getOuterCommandType();

    void setOuterCommandType(String type);

}
